src/pages/resultsView/ResultsViewPageHelpers.ts

```diff
@@ -0,0 +1,141 @@
+        }
+
+        if (query.data_priority !== undefined && parseInt(query.data_priority,10) !== resultsViewPageStore.profileFilter) {
+            resultsViewPageStore.profileFilter = parseInt(query.data_priority,10);
+        }
+
+        // note that this could be zero length if we have multiple studies
+        // in that case we derive default selected profiles
+        const profiles = getMolecularProfiles(query);
+        if (!resultsViewPageStore.selectedMolecularProfileIds || !_.isEqual(resultsViewPageStore.selectedMolecularProfileIds.slice(), profiles)) {
+            resultsViewPageStore.selectedMolecularProfileIds = profiles;
+        }
+
+        if (!_.isEqual(query.RPPA_SCORE_THRESHOLD, resultsViewPageStore.rppaScoreThreshold)) {
+            resultsViewPageStore.rppaScoreThreshold = parseFloat(query.RPPA_SCORE_THRESHOLD);
+        }
+
+        if (!_.isEqual(query.Z_SCORE_THRESHOLD, resultsViewPageStore.zScoreThreshold)) {
+            resultsViewPageStore.zScoreThreshold = parseFloat(query.Z_SCORE_THRESHOLD);
+        }
+
+        if (query.geneset_list) {
+            // we have to trim because for some reason we get a single space from submission
+            const parsedGeneSetList = query.geneset_list.trim().length ? (query.geneset_list.trim().split(/\s+/)) : [];
+            if (!_.isEqual(parsedGeneSetList, resultsViewPageStore.genesetIds)) {
+                resultsViewPageStore.genesetIds = parsedGeneSetList;
+            }
+        }
+
+        // cohortIdsList will contain virtual study ids (physicalstudies will contain the phsyical studies which comprise the virtual studies)
+        // although resultsViewStore doesn
+        if (!resultsViewPageStore.cohortIdsList || !_.isEqual(resultsViewPageStore.cohortIdsList.slice(), cohortIdsList)) {
+            resultsViewPageStore.cohortIdsList = cohortIdsList;
+        }
+
+        if (resultsViewPageStore.oqlQuery !== oql) {
+            resultsViewPageStore.oqlQuery = oql;
+        }
+
+}
+
+export function getVirtualStudies(cancerStudyIds:string[]):Promise<VirtualStudy[]>{
+
+    const prom = new Promise<VirtualStudy[]>((resolve, reject)=>{
+        client.getAllStudiesUsingGET({projection:"SUMMARY"}).then((allStudies)=>{
+            //console.log(cancerStudyIds);
+            //console.log(allStudies);
+            const virtualStudyIds = _.differenceWith(cancerStudyIds, allStudies,(id:string, study:CancerStudy)=>id==study.studyId);
+
+            if (virtualStudyIds.length > 0) {
+                Promise.all(virtualStudyIds.map(id =>  sessionServiceClient.getVirtualStudy(id)))
+                    .then((virtualStudies)=>{
+                         resolve(virtualStudies);
+                    })
+            } else {
+                resolve([]);
+            }
+        });
+    });
+    return prom;
+
+}
+
+export function substitutePhysicalStudiesForVirtualStudies(cancerStudyIds:string[], virtualStudies:VirtualStudy[]){
```

Comment:
just so Im clear on terminology - what constitutes a virtual study? is a multiple-study query a virtual study? or is a virtual study something the user has *saved*. and is there any other kind of study besides physical and virtual?

Reply:
A virtual study is one that has been saved using the vs service.   It has  a study ID that looks like a guid and we use the vs service to retrieve it's properties.  Right now, it's just a set of studies and lists of samples in those studies.

Reply:
got it thanks